roles/v2/pool/src/status.rs

```diff
@@ -0,0 +1,115 @@
+use crate::error::PoolError;
+
+/// Each sending side of the status channel 
+/// should be wrapped with this enum to allow 
+/// the main thread to know which component sent the message
+#[derive(Debug)]
+pub enum Sender {
+    Downstream(async_channel::Sender<Status>),
+    DownstreamListener(async_channel::Sender<Status>),
+    Upstream(async_channel::Sender<Status>),
+}
+
+impl Sender {
+    /// used to clone the sending side of the status channel used by the TCP Listener
+    /// into individual Sender's for each Downstream instance
+    pub fn listener_to_connection(&self) -> Self {
+        match self {
+            Self::DownstreamListener(inner) => Self::Downstream(inner.clone()),
+            _ => unreachable!(),
```

Comment:
I'm probably missing context on what makes this unreachable, but should this be a different panic of some kind, describing the issue?

Reply:
Possibly. For context, the `Sender<Status>` is wrapped in an enum to let the main thread know which component encountered the error. This function is used to clone the `Sender` sent to the TCP listener component on creation into a new `Sender` for each `Downstream` that gets created by the listener. Since this is the only time there is a Parent -> Child relationship regarding the `Sender<Status>` type, it is only needed for this specific use case.

Reply:
Might be good to at least document in the doc comment that it panics if not called on a Self::DownstreamListener. This feels weird to me, but I think I'm missing too much context still to suggest an alternative at this time.